tools/classes/PresenceCompiler.ts

```diff
@@ -0,0 +1,285 @@
+											":" +
+											chalk.yellowBright(error.line) +
+											":" +
+											chalk.yellowBright(error.character) +
+											" - " +
+											chalk.redBright("Error ") +
+											chalk.gray("TS" + error.code + ": ") +
+											error.content
+									)
+								);
+							},
+						},
+					},
+				],
+			},
+			...(this.options?.webpack || {}),
+		};
+
+		if (!options.transpileOnly) options.transpileOnly = false;
+
+		if (Array.isArray(presence)) {
+			let errors: webpack.WebpackError[] = [];
+
+			actions.info(chalk.yellow(`Compiling ${presence.length} Presence(s)`));
+			for (const p of presence) {
+				const presencePath = this.getPresenceFolder(p);
+
+				await writeFile(resolve(presencePath, "tsconfig.json"), tsconfig);
+				await this.installPresenceDependencies(p);
+
+				const job = await new Promise<{
+					error: Error | undefined;
+					stats: webpack.Stats | undefined;
+				}>(r =>
+					webpack.webpack(
+						{
+							...webpackConfig,
+							context: presencePath,
+							output: options.noEmit
+								? undefined
+								: {
+										iife: false,
+										path: presencePath,
+										filename: "[name].js",
+								  },
+							entry: {
+								presence: "./presence.ts",
+								...(existsSync(resolve(presencePath, "iframe.ts"))
+									? { iframe: "./iframe.ts" }
+									: {}),
+							},
+						},
+						(error, stats) => r({ error, stats })
+					)
+				);
+
+				if (job.error) throw job.error;
+
+				errors.push(...(job.stats?.compilation?.errors || []));
+			}
+
+			for (const p of presence)
+				if (existsSync(resolve(this.getPresenceFolder(p), "tsconfig.json")))
+					await rm(resolve(this.getPresenceFolder(p), "tsconfig.json"));
+
+			errors = errors.filter(e => e.name !== "ModuleBuildError");
+
+			if (!errors.length) {
+				if (!options.transpileOnly)
+					actions.info(
+						chalk.green(`Successfully compiled ${presence.length} Presence(s)!`)
+					);
+				else
+					actions.info(
+						chalk.green(
+							`Successfully transpiled ${presence.length} Presence(s)!`
+						)
+					);
+			}
+
+			return errors;
+		}
+
+		const presencePath = this.getPresenceFolder(presence);
+
+		if (!options.output) options.output = presencePath;
+
+		await writeFile(resolve(presencePath, "tsconfig.json"), tsconfig);
+
+		await this.installPresenceDependencies(presence);
+
+		actions.info(chalk.yellow(`Compiling ${presence}...`));
+		const job = await new Promise<{
+			err: Error | undefined;
+			stats: webpack.Stats | undefined;
+		}>(r => {
+			return webpack.webpack(
+				{
+					...webpackConfig,
+					context: presencePath,
+					entry: {
+						presence: "./presence.ts",
+						...(existsSync(resolve(presencePath, "iframe.ts"))
+							? { iframe: "./iframe.ts" }
+							: {}),
+					},
+				},
+				(err, stats) => r({ err, stats })
+			);
+		});
+
+		if (existsSync(resolve(presencePath, "tsconfig.json")))
+			await rm(resolve(presencePath, "tsconfig.json"));
```

Comment:
why this and L141

Reply:
> why this and L141

That one is for Arrays, this one is for singular